### ISSUE TITLE
PHP8.2 support ?

### ISSUE THREAD

**Issue by Nuranto** [owebia/magento2-module-advanced-shipping#113]
Hi,

Since latest Magento (2.4.6) package now supports PHP8.2, we should add support to this package too.

**Comment by Nuranto** [comment]
Thanks @owebia 

A major release, should we expect some breaking changes in 6.0.0 ?

**Comment by owebia** [comment]
It's mainly a code refactor.

Here is the current changelog:
https://owebia.com/static/owebia/magento2-module-advanced-shipping/CHANGELOG.md

In the breaking changes:
- remove support for PHP < 7.4
- internal classes refactored (for those who extended them or where overriding them in di.xml)

There may be other breaking changes added (remove support for old Magento versions for example).

**Comment by Nuranto** [comment]
Sounds good! Thanks!

**Comment by iphigenie** [comment]
installing the betas on our 2.4.6 testing, how can we send in the errors we find, for information?

**Comment by owebia** [comment]
Hi @iphigenie ,

Sorry for the late reply.
Could you please send us by email at contact at owebia.com your configuration so that we can try to reproduce your issue?

Thanks!

**Comment by iphigenie** [comment]
Hi @owebia - have done so

But not too hard to explain:

It's conditions like (preg_match('/^(BD|LS19|LS11|LS23).*$/i', $request->dest_postcode))

shopping cart "estimate shipping" or full checkout

address still empty so the fields it is using for conditions are empty

some of these must therefore receive a NULL and that's not longer tolerated.

**Comment by owebia** [comment]
Thanks for your answer.

About your issue, you should write something like this:
```php
preg_match('/^(BD|LS19|LS11|LS23).*$/i', $request->dest_postcode ?? '')
```

It would not be appropriate to change native functions behavior and it's Magento that provides the data.
We can update the documentation to suggest using `?? ''` when required.

Is there another issue with preg_match conditions (the first you reported)?

**Comment by iphigenie** [comment]
I'm not sure there was another issue. 
I had indeed blindly gone by the documentation, should have figured that out.

**Comment by lukaszwoznica** [comment]
Hi @owebia
Although a new version with support for PHP 8.2 has been released, the dependency has not been updated in composer.json.

**Comment by owebia** [comment]
Version `6.0.1` released with PHP 8.2 support
